packages/orchestration/src/exos/chain-hub.js

```diff
@@ -425,6 +437,20 @@ export const makeChainHub = (zone, agoricNames, vowTools) => {
       }
       return undefined;
     },
+    /**
+     * @param {string} address bech32 address
+     * @returns {CosmosChainInfo}
+     */
+    getChainInfoFromAddress(address) {
+      const [prefix, rest] = address.split('1');
+      (prefix && rest) ||
+        Fail`Address contains invalid bech32 separator: ${q(address)}`;
```

Comment:
cosmjs uses [bech32](https://www.npmjs.com/package/bech32), which uses `lastIndexOf`, not the 1st occurrence.

https://github.com/bitcoinjs/bech32/blob/5ceb0e3d4625561a459c85643ca6947739b2d83c/src/index.ts#L146

I wonder about factoring out a `getBech32Prefix(addr)` with its own unit tests.

Reply:
the so-called reference implementation uses `.rfind('1')` too.

https://github.com/sipa/bech32/blob/7a7d7ab158db7078a333384e0e918c90dbc42917/ref/python/segwit_addr.py#L79

Reply:
Thanks  - glad I tagged you as a reviewer! Will factor this out to its own function and adopt the new logic.

Reply:
Interestingly, in practice no one seems to use `1` in the prefix / human readable part:
https://github.com/satoshilabs/slips/blob/master/slip-0173.md#registered-human-readable-parts

But the spec **_does say_** any US-ASCII character is valid for the HRP:
https://github.com/bitcoin/bips/blob/7a104491862379e88662a39d1a2b613c907e503e/bip-0173.mediawiki#bech32

Please see the amended commit: https://github.com/Agoric/agoric-sdk/pull/10354/commits/65e07f8db405c6c473ad62a64c2cb6f2166da159